### PR TITLE
Update golangci-lint to v1.45.2

### PR DESCRIPTION
**Pull request by Sean-Der**
Also enable more linters and remove deprecated ones

### CODE REVIEW

File: ci/.golangci.yml [hunk 1]
```diff
@@ -35,40 +43,62 @@ linters:
...
     - unparam          # Reports unused function parameters
     - unused           # Checks Go code for unused constants, variables, functions and types
     - varcheck         # Finds unused global variables and constants
+    - wastedassign     # wastedassign finds wasted assignment statements
     - whitespace       # Tool for detection of leading and trailing whitespace
   disable:
+    - containedctx     # containedctx is a linter that detects struct contained context.Context field
+    - cyclop           # checks function and package cyclomatic complexity
+    - exhaustivestruct # Checks if all struct's fields are initialized
+    - forbidigo        # Forbids identifiers
```
Comment (Sean-Der) [inline ci/.golangci.yml:79]: I will enable this in a follow up commit. Replaces `lint-disallowed-functions-in-library.sh`

File: ci/.golangci.yml [hunk 2]
```diff
@@ -35,40 +43,62 @@ linters:
...
+    - ifshort          # Checks that your code uses short syntax for if-statements whenever possible
+    - ireturn          # Accept Interfaces, Return Concrete Types
     - lll              # Reports long lines
+    - maintidx         # maintidx measures the maintainability index of each function.
+    - makezero         # Finds slice declarations with non-zero initial length
     - maligned         # Tool to detect Go structs that would take less memory if their fields were sorted
     - nestif           # Reports deeply nested if statements
     - nlreturn         # nlreturn checks for a new line before return and branch statements to increase code clarity
     - nolintlint       # Reports ill-formed or insufficient nolint directives
+    - paralleltest     # paralleltest detects missing usage of t.Parallel() method in your Go test
```
Comment (Sean-Der) [inline ci/.golangci.yml:93]: I will enable this in a follow up commit. I am excited this could reduce our test time alot!

**Comment by Sean-Der** [comment]
Not a real exciting PR! If anyone agrees/disagrees with the configuration happy to change it.

I tested this configuration on pion/webrtc and it caught some bugs (around checking and returning errors) 

I disabled linters because it would be too hard to fix them all today. I am all for enabling in the future though.

**Comment by Sean-Der** [comment]
I just added active comitters across the org, please ignore if this isn't interesting to you :)

File: ci/.golangci.yml (hunk 2 above)
Comment (kylecarbs) [reply ci/.golangci.yml:93]: We use `paralleltest` @coder and it's helped enforcement a lot!

**Review by boks1971 — APPROVED** [review]
This is great! I don't understand all the settings, but more automated/static checks are always good :-)

**Review by kylecarbs — APPROVED** [review]
Seems reasonable to me!

**Review by r-novel — APPROVED** [review]

**Review by enobufs — APPROVED** [review]
Looks good to me. Thnx!

**Review by jkralik — APPROVED** [review]

**Review by davidzhao — APPROVED** [review]
lgtm!

**Pull request merged**